provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,251 @@
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_017: [ If the registration response has a success code with a 'status' of 'Assigning', `register` shall start polling for operation updates ] */
+                  this._operationId = responseBody.operationId;
+                  this._fsm.transition('waiting_to_poll', callback);
+                } else if (responseBody.status.toLowerCase() === 'assigned') {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_015: [ If the registration response has a success code with a 'status' of 'Assigned', `register` call the `callback` with `err` == `null` and result `containing` the deserialized body ] */
+                  this._fsm.transition('idle', callback, null, responseBody);
+                } else {
+                  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_029: [ If the registration response has a success code with a 'status' that is any other value', `register` shall call the callback with a `SyntaxError` error. ] */
+                  this._fsm.transition('idle', callback, new SyntaxError('badly formed registration response'));
+                }
+              }
+            });
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle', callback, err);
+            });
+          }
+        },
+        waiting_to_poll: {
+          _onEnter: (callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_018: [ `register` shall poll for operation status every `operationStatusPollingInterval` milliseconds ] */
+            this._pollingTimer = setTimeout(() => {
+              this._fsm.transition('polling',  callback);
+            }, this._operationStatusPollingInterval);
+
+          },
+          register: (authorization, forceRegistration, body, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_HTTP_18_036: [ `register` shall call the `callback` with an `InvalidOperationError` if it is called while a previous registration is in progress. ] */
+            callback(new errors.InvalidOperationError('registration already in progress.'));
+          },
+          disconnect : (callback) => {
+            this._cancelCurrentOperation((err) => {
+              this._fsm.transition('idle', callback, err);
+            });
+          }
+        },
+        polling: {
+          _onEnter: (callback) => {
+            this._sendOperationStatusRequest((err, body) => {
+              if (this._registrationInProgress()) { // make sure we weren't cancelled before doing something with the response
```

Comment:
>if (this._registrationInProgress()) [](start = 14, length = 35)

for these guys I believe it'd be nice to have an else clause with a debug message. generally speaking it'd be useful to add more debug() traces everywhere.